app/controllers/admin/people_controller.rb

```diff
@@ -3,5 +3,32 @@ class PeopleController < AdminController
     def index
       @people = Person.all
     end
+
+    def batch_update
+      changes_made = false
+      people_params.each do |param|
+        person = Person.find(param[:id])
```

Comment:
We have all the person ids at the start, any way we can preload all these records? 
Remember that in production each db request can take up to 200ms.

Another performance improvement we could bring here would be to wrap this whole loop with an ActiveRecord transaction (ill let you read up on that)